#!/usr/bin/env python
PACKAGE = "auv_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Position weights
gen.add("w_pos_x", double_t, 0, "Weight for position X", 100.0, 0.0, 2000.0)
gen.add("w_pos_y", double_t, 0, "Weight for position Y", 100.0, 0.0, 2000.0)
gen.add("w_pos_z", double_t, 0, "Weight for position Z", 400.0, 0.0, 2000.0)

# Orientation weight (scalar)
gen.add("w_ori", double_t, 0, "Weight for orientation", 150.0, 0.0, 5000.0)

# Velocity weights
gen.add("w_vel_x", double_t, 0, "Weight for linear velocity X", 11.0, 0.0, 1000.0)
gen.add("w_vel_y", double_t, 0, "Weight for linear velocity Y", 11.0, 0.0, 1000.0)
gen.add("w_vel_z", double_t, 0, "Weight for linear velocity Z", 11.0, 0.0, 1000.0)
gen.add("w_vel_roll", double_t, 0, "Weight for angular velocity Roll", 11.0, 0.0, 1000.0)
gen.add("w_vel_pitch", double_t, 0, "Weight for angular velocity Pitch", 11.0, 0.0, 1000.0)
gen.add("w_vel_yaw", double_t, 0, "Weight for angular velocity Yaw", 11.0, 0.0, 1000.0)

# Control input weights
gen.add("w_tau_x", double_t, 0, "Weight for force X", 0.03, 0.0, 100.0)
gen.add("w_tau_y", double_t, 0, "Weight for force Y", 0.03, 0.0, 100.0)
gen.add("w_tau_z", double_t, 0, "Weight for force Z", 0.05, 0.0, 100.0)
gen.add("w_tau_roll", double_t, 0, "Weight for torque Roll", 0.02, 0.0, 100.0)
gen.add("w_tau_pitch", double_t, 0, "Weight for torque Pitch", 0.02, 0.0, 100.0)
gen.add("w_tau_yaw", double_t, 0, "Weight for torque Yaw", 0.02, 0.0, 100.0)

# Terminal Position weights
gen.add("wN_pos_x", double_t, 0, "Terminal Weight for position X", 500.0, 0.0, 5000.0)
gen.add("wN_pos_y", double_t, 0, "Terminal Weight for position Y", 500.0, 0.0, 5000.0)
gen.add("wN_pos_z", double_t, 0, "Terminal Weight for position Z", 2000.0, 0.0, 5000.0)

# Terminal Orientation weight (scalar)
gen.add("wN_ori", double_t, 0, "Terminal Weight for orientation", 600.0, 0.0, 10000.0)

# Terminal Velocity weights
gen.add("wN_vel_x", double_t, 0, "Terminal Weight for linear velocity X", 55.0, 0.0, 1000.0)
gen.add("wN_vel_y", double_t, 0, "Terminal Weight for linear velocity Y", 55.0, 0.0, 1000.0)
gen.add("wN_vel_z", double_t, 0, "Terminal Weight for linear velocity Z", 55.0, 0.0, 1000.0)
gen.add("wN_vel_roll", double_t, 0, "Terminal Weight for angular velocity Roll", 55.0, 0.0, 1000.0)
gen.add("wN_vel_pitch", double_t, 0, "Terminal Weight for angular velocity Pitch", 55.0, 0.0, 1000.0)
gen.add("wN_vel_yaw", double_t, 0, "Terminal Weight for angular velocity Yaw", 55.0, 0.0, 1000.0)

exit(gen.generate(PACKAGE, "auv_control", "MPCWeights"))
